dot/state/grandpa.go

```diff
@@ -24,44 +26,570 @@ var (
 	currentSetIDKey   = []byte("setID")
 )
 
+var (
```

Comment:
all of this pendingChange related code should probably go in its own file

Reply:
Agreed!